src/network/clientpackethandler.cpp

```diff
@@ -982,6 +982,22 @@ void Client::handleCommand_SpawnParticle(NetworkPacket* pkt)
 	m_client_event_queue.push(event);
 }
 
+/* this function should not exist. alas, i managed to leave out critical fields
+ * in the original tween serialization/deserialization code and now the protocol
+ * is fixed, so now we all have to suffer for my abject stupidity. j o y */
+template <typename T>
+inline bool readMissingTweenParamsFor(std::istringstream& is, ParticleParamTypes::TweenedParameter<T>& t) {
+	u8 tmp;
+	tmp = readU8(is);
+	if (is.eof())
+		return false;
+
+	t.style = (decltype(t.style))tmp;
```

Comment:
I suppose this is more C++-y
```suggestion
	t.style = static_cast<decltype(t.style)>(tmp);
```